crates/primitives/src/transaction/compat.rs

```diff
@@ -12,8 +12,8 @@ pub trait FillTxEnv {
 
 impl FillTxEnv for TransactionSigned {
     fn fill_tx_env(&self, tx_env: &mut TxEnv, sender: Address) {
-        #[cfg(all(feature = "optimism", not(feature = "scroll")))]
-        let envelope = alloy_eips::eip2718::Encodable2718::encoded_2718(self);
+        #[cfg(any(feature = "optimism", feature = "scroll"))]
+        let _envelope = alloy_eips::eip2718::Encodable2718::encoded_2718(self);
```

Comment:
Did we add the `_` because of failing lints? Possibly the correct feature gate would be:
```rust
        #[cfg(any(
            all(feature = "optimism", not(feature = "scroll")),
            all(feature = "scroll", not(feature = "optimism"))
        ))]
```